src/pages/workspaces/workspace/tools/WorkflowView.js

```diff
@@ -147,10 +149,12 @@ class WorkflowView extends Component {
   }
 
   componentDidUpdate() {
-    const { selectedTabIndex, loadedWdl } = this.state
+    const { config, entityTypes, inputsOutputs, invalid, modifiedAttributes, selectedTabIndex, loadedWdl, wdl } = this.state
     if (selectedTabIndex === 2 && !loadedWdl) {
       this.fetchWDL()
     }
+
+    StateHistory.update({ config, entityTypes, inputsOutputs, invalid, modifiedAttributes, selectedTabIndex, wdl })
```

Comment:
This seems like a lot of data to be caching, especially since some of it is derived. Given that each piece of data we store increases the chance of state-caching related bugs down the line, could we get away with less and still have a reasonable UX?

Reply:
I think this is what we need.